### PR TITLE
Material plugins to Webgpu (+meshUVSpaceRenderer)

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by deltakosh** [comment]
Material plugin doc was also updated accordingly

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15409/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15409/merge/testResults/webgpuplaywright/index.html

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/ShadersWGSL/meshUVSpaceRendererFinaliser.fragment.fx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+    var mask: vec4f = textureSample(maskTextureSampler, maskTextureSamplerSampler, input.vUV).rgba;
+
+    if (mask.r > 0.5) {
+        fragmentOutputs.color = textureSample(textureSampler, textureSamplerSampler, input.vUV);
+    } else {
+        var texelSize: vec2f = 4.0 / uniforms.textureSize;
+
+        var uv_p01: vec2f = vUV +  vec2f(-1.0, 0.0) * texelSize;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/meshUVSpaceRendererFinaliser.fragment.fx:20]:
```suggestion
        var uv_p01: vec2f = input.vUV +  vec2f(-1.0, 0.0) * texelSize;
```

File: packages/dev/core/src/Materials/PBR/pbrAnisotropicConfiguration.ts [hunk 2]
```diff
@@ -105,6 +105,14 @@ export class PBRAnisotropicConfiguration extends MaterialPluginBase {
         this._internalMarkAllSubMeshesAsMiscDirty();
     }
 
+    /**
+     * Gets a boolean indicating that the plugin is compatible with a give shader language.
```
Comment (sebavan) [inline packages/dev/core/src/Materials/PBR/pbrAnisotropicConfiguration.ts:109]: "with a give shader language" sounds french or almost :-)

File: packages/dev/core/src/ShadersWGSL/meshUVSpaceRendererFinaliser.vertex.fx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+// Uniforms
+uniform worldViewProjection: mat4x4f;
+
+// Varyings
+varying vUV: vec2f;
+
+@vertex
+fn main(input : VertexInputs) -> FragmentInputs {
+    vertexOutpus.position = uniforms.worldViewProjection *  vec4f(input.position, 1.0);
+    vertexOutpus.positionvUV = uv;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/meshUVSpaceRendererFinaliser.vertex.fx:14]:
```suggestion
    vertexOutputs.position = uniforms.worldViewProjection *  vec4f(input.position, 1.0);
    vertexOutputs.vUV = input.uv;
```

File: packages/dev/core/src/ShadersWGSL/meshUVSpaceRendererMasker.vertex.fx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+attribute uv: vec2f;
+
+varying vUV: vec2f;
+
+@vertex
+fn main(input : VertexInputs) -> FragmentInputs {
+    vertexOutpus.position =  vec4f( vec2f(input.uv.x, input.uv.y) * 2.0 - 1.0, 0., 1.0);
+    vertexOutpus.vUV = uv;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/meshUVSpaceRendererMasker.vertex.fx:8]:
```suggestion
    vertexOutputs.position =  vec4f( vec2f(input.uv.x, input.uv.y) * 2.0 - 1.0, 0., 1.0);
    vertexOutputs.vUV = input.uv;
```

**Review by noname0310 — COMMENTED** [review]
Maybe the shader isn't in the test. These are trivial syntax errors

File: packages/dev/core/src/Materials/meshDebugPluginMaterial.ts [hunk 5]
```diff
@@ -103,6 +148,52 @@ flat varying float dbg_vPass;
...
 
+const fragmentDefinitionsWebGPU = `#if defined(DBG_ENABLED)
+varying dbg_vBarycentric: vec3f;
+varying dbg_vVertexWorldPos: vec3f;
+varying dbg_vPass: f32;
+
+#if !defined(DBG_MULTIPLY)
+    fn dbg_applyShading(color: vec3f) -> vec3f {
+        var N = input.vNormalW.xyz;
+        var L = normalize(scene.vEyePosition.xyz - input.vPositionW.xyz);
```
Comment (noname0310) [inline packages/dev/core/src/Materials/meshDebugPluginMaterial.ts:159]: variable `input` is out of scope in this context
use fragmentInputs instead

Comment (deltakosh) [reply packages/dev/core/src/Materials/meshDebugPluginMaterial.ts:159]: nice ones! thanks I will fix that !

File: packages/dev/core/src/Materials/meshDebugPluginMaterial.ts [hunk 6]
```diff
@@ -135,6 +226,38 @@ vec3 dbg_color = vec3(1.);
 #endif
 #endif`;
 
+const fragmentMainEndWebGPU = `#if defined(DBG_ENABLED)
+var dbg_color = vec3f(1.);
+#if DBG_MODE == 1
+    dbg_color = mix(uniforms.dbg_wireframeTrianglesColor, vec3f(1.), dbg_edgeFactor());
+#elif DBG_MODE == 2 || DBG_MODE == 3
+    var dbg_cornerFactor = dbg_cornerFactor();
+    if (dbg_vPass == 0. && dbg_cornerFactor == 1.) discard;
```
Comment (noname0310) [inline packages/dev/core/src/Materials/meshDebugPluginMaterial.ts:235]: `fragmentInputs.dbg_vPass`

Comment (deltakosh) [reply packages/dev/core/src/Materials/meshDebugPluginMaterial.ts:235]: I'll add a test

**Pull request merged**